test/e2e/sparkapplication_test.go

```diff
@@ -236,6 +237,121 @@ var _ = Describe("Example SparkApplication", func() {
 		})
 	})
 
+	Context("fail-submission", func() {
+		ctx := context.Background()
+		path := filepath.Join("bad_examples", "fail-submission.yaml")
+		app := &v1beta2.SparkApplication{}
+
+		BeforeEach(func() {
+			By("Parsing SparkApplication from file")
+			file, err := os.Open(path)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(file).NotTo(BeNil())
+
+			decoder := yaml.NewYAMLOrJSONDecoder(file, 100)
+			Expect(decoder).NotTo(BeNil())
+			Expect(decoder.Decode(app)).NotTo(HaveOccurred())
+
+			By("Creating SparkApplication")
+			Expect(k8sClient.Create(ctx, app)).To(Succeed())
+		})
+
+		AfterEach(func() {
+			key := types.NamespacedName{Namespace: app.Namespace, Name: app.Name}
+			Expect(k8sClient.Get(ctx, key, app)).To(Succeed())
+
+			By("Deleting SparkApplication")
+			Expect(k8sClient.Delete(ctx, app)).To(Succeed())
+		})
+
+		It("Fails submission and retries until retries are exhausted", func() {
+			By("Waiting for SparkApplication to terminate")
+			key := types.NamespacedName{Namespace: app.Namespace, Name: app.Name}
+			apps, polling_err := collectSparkApplicationsUntilTermination(ctx, key)
+			Expect(polling_err).To(HaveOccurred())
+
+			By("Should eventually fail")
+			final_app := apps[len(apps)-1]
+			Expect(final_app.Status.AppState.State).To(Equal(v1beta2.ApplicationStateFailed))
+			Expect(final_app.Status.AppState.ErrorMessage).To(ContainSubstring("failed to run spark-submit"))
+			Expect(final_app.Status.SubmissionAttempts).To(Equal(*app.Spec.RestartPolicy.OnSubmissionFailureRetries + 1))
```

Comment:
In golang, we prefer using the `camelCase` naming convention instead of using an underscore. Ref: https://go.dev/doc/effective_go#mixed-caps.